lib/packages/docutils/rst.nim

```diff
@@ -1018,15 +1036,43 @@ proc fixupEmbeddedRef(n, a, b: PRstNode) =
   for i in countup(0, sep - incr): a.add(n.sons[i])
   for i in countup(sep + 1, n.len - 2): b.add(n.sons[i])
 
-proc whichRole(sym: string): RstNodeKind =
-  case sym
+const supportedLanguages = ["nim", "yaml", "python", "java", "c",
+                            "c++", "cpp", "c#", "csharp"]
+proc whichRole(p: RstParser, sym: string): RstNodeKind =
+  let r = sym.toLowerAscii
+  case r
   of "idx": result = rnIdx
   of "literal": result = rnInlineLiteral
   of "strong": result = rnStrongEmphasis
   of "emphasis": result = rnEmphasis
   of "sub", "subscript": result = rnSub
   of "sup", "superscript": result = rnSup
-  else: result = rnGeneralRole
+  # literal and code are the same in our implementation
+  of "code": result = rnInlineLiteral
+  # c++ currently can be spelled only as cpp, c# only as csharp
+  elif r in supportedLanguages:
+    result = rnInlineCode
+  else:  # unknown role
+    rstMessage(p, mwUnsupportedLanguage, p.s.currRole)
+    result = rnGeneralRole
+
+proc toInlineCode(n: PRstNode, language: string): PRstNode =
+  ## Creates rnInlineCode and attaches `n` contents as code (in 3rd son).
+  result = newRstNode(rnInlineCode)
+  var args = newRstNode(rnDirArg)
+  var lang = language
+  if language == "cpp": lang = "c++"
```

Comment:
something is inconsistent:
```
.. code-block:: cpp

  #include <stdio.h>

.. code-block:: c++

  #include <stdio.h>

`#include <stdio.h>`:cpp:

`#include <stdio.h>`:c++:
```

gives 
![image](https://user-images.githubusercontent.com/2194784/113374086-2215f400-9321-11eb-8920-6617bddbdfae.png)

=> depending on block vs inline, the syntax that works is c++ vs cpp

we should have one that works in all contexts, and recommend it (including in example from eariler).

Reply:
The fact that `:c++:` is not parsed is another bug in our parser. Added to the list of minor bugs.

The proper fix (with code reusing) is not trivial, it's better to handle in another PR.